jetty-ee9/jetty-ee9-servlet/src/test/java/org/eclipse/jetty/ee9/servlet/HttpChannelEventTest.java

```diff
@@ -224,32 +229,29 @@ public void onComplete(Request request)
         List<HttpField> xheaders = response.getFields("X-Header");
         assertThat("X-Header count", xheaders.size(), is(1));
         assertThat("X-Header[0].value", xheaders.get(0).getValue(), is("zed"));
-             */
     }
 
     @Test
     public void testResponseFailure() throws Exception
     {
-        /* TODO
-        start(new TestHandler()
+        start(new TestServlet()
         {
             @Override
-            protected void handle(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException
+            protected void handle(HttpServletRequest request, HttpServletResponse response) throws IOException
             {
                 // Closes all connections, response will fail.
                 connector.getConnectedEndPoints().forEach(EndPoint::close);
             }
         });
 
-         */
-
         CountDownLatch latch = new CountDownLatch(2);
-        /* TODO
+
         connector.addBean(new HttpChannel.Listener()
         {
             @Override
             public void onResponseFailure(Request request, Throwable failure)
             {
+                //TODO this is never called
```

Comment:
This could be a bug.  I see no reason that a response failure within a context cannot be notified like this.
Further analysis required.